gr-zeromq/include/gnuradio/zeromq/pub_sink.h

```diff
@@ -25,7 +25,11 @@ namespace zeromq {
  * This block acts a a streaming sink for a GNU Radio flowgraph
  * and writes its contents to a ZMQ PUB socket.  A PUB socket may
  * have subscribers and will pass all incoming stream data to each
- * subscriber.  Subscribers can be either another gr-zeromq source
+ * subscriber with a matching key. If the publisher's key is set to
+ * "GNURadio", the following example subscriber keys will match: "G",
+ * "GN", .., "GNURadio", and "". In other words, the subscriber must
```

Comment:
I don't think "" will work here. As noted here & in the code, the key size has to be > 0, so all the rest might work but I don't think "" will. Yes? If so, please correct the comment.

Reply:
Yes, I'll modify the docs to make it clearer.

The empty string is a bit of special case: If a publisher inserts a key, but the subscriber sets the key to an empty string, the key will end up being output as data on the subscriber.
Sometimes this will give garbage data, but it's more likely that the exception in gr-zeromq/lib/base_impl.cc:245 will be thrown because key.size() will not be a multiple of sizeof(output_data_type).